app/Dockerfile

```diff
@@ -1,44 +1,40 @@
-FROM alpine:3.8
-
-# install python and pip with apk package manager
-RUN apk -Uuv add python py-pip
+FROM python:3.8-slim AS builder
 
 # uwsgi in particular requires a lot of packages to install, delete them afterwards
-RUN apk add --no-cache \
+RUN apt-get update && apt-get -y --no-install-recommends install \
         gettext \
-        python3 \
-        build-base \
-        linux-headers \
-        python3-dev
+        gcc \
+        g++ \
+        make \
+        libc6-dev \
+    && apt-get clean
 
 # copies requirements.txt to the docker container
 ADD requirements.txt requirements.txt
 
-# install requirements
-RUN pip3 install --upgrade pip
-RUN pip3 install --no-cache-dir -r requirements.txt
-
-# clean up environment by deleting extra packages
-RUN apk del \
-        build-base \
-        linux-headers \
-        python3-dev
+# Install requirements
+RUN pip3 install --user --no-cache-dir -r requirements.txt
 
-# specify the directory that CMD executes from
-WORKDIR /home
-
-# copy over the directory into docker container with given path
-COPY . /home
+FROM python:3.8-slim
 
-# create unprivileged user account
-RUN addgroup aladdin-user && \
-    adduser -SD -G aladdin-user aladdin-user
+# Create unprivileged user account
+RUN useradd -m -U -d /home/aladdin-user aladdin-user
 
 # switch to the unpriveleged user account
 USER aladdin-user
 
 # display how old the python libraries are
 RUN pip3 list --no-cache-dir --outdated
```

Comment:
move this to after copying dependencies

Reply:
Why?

Reply:
well the purpose of this is to list outdated dependencies, right? so I'm assuming this will always be empty since this container does not yet have any dependencies installed, maybe I'm missing something

Reply:
... because I can't read.  Yes, taken care of.